internal/retry/command.go

```diff
@@ -54,7 +54,10 @@ func (c commandDoer) DoWithErrorHandling(ctx context.Context, errorFunc ErrorFun
 	return retry.Do(func() error {
 		cmd, stdout, stderr := c.createCmd(ctx)
 		err := cmd.Run()
-		return errorFunc(err, stdout, stderr)
+		if err != nil {
+			return errorFunc(err, stdout, stderr)
+		}
+		return nil
```

Comment:
I'd leave this as is, always passing `err`, not only when it's `nil`.

Reply:
IMO `DoWithErrorHandling` name suggests that the `ErrorFunc` it gets will be called when an error appears, not when it's nil. If it was to handle every result, it should be named differently, e.g. `DoWithResultHandling`.

Reply:
🤔 Maybe you're right. I omitted that somehow. We might add a comment to `DoWithErrorHandling` that `err` is never `nil`.

Reply:
PTAL: 1eb4a7c